mediorum/server/serve_upload.go

```diff
@@ -106,20 +108,18 @@ func (ss *MediorumServer) postUpload(c echo.Context) error {
 	return c.JSON(200, uploads)
 }
 
-func hashFileUpload(upload *multipart.FileHeader) (string, error) {
-	// for testing... want to be able to upload same stuff repeatedly
-	return randomHash()
-
-	file, err := upload.Open()
+func computeFileCID(file io.Reader) (string, error) {
+	builder := cid.V1Builder{}
+	contents, err := ioutil.ReadAll(file)
```

Comment:
Two problems with using `ReadAll` here:
* it consumes the reader passed in... so calling code needs to `Seek(0,0)` if it expects reader to still work. 
* it reads whole file into memory, which for a wav could easily mean 100+ MB.

Anyway I think you can solve both by:
* making param ReedSeeker
* removing `ReadAll` and using [CidFromReader](https://pkg.go.dev/github.com/ipfs/go-cid#CidFromReader)
* doing a `defer file.Seek(0,0)`

next week planning to do the "parallel replicate" change, which will pass around files everywhere so we can have independent readers which will improve this.

Reply:
for not using `ReadAll`, something like:

```
h, err := mh.SumStream(file, DefaultIpfsHash, -1)
```

basically modify the [utils](https://github.com/ipfs/go-ipfs-util/blob/master/util.go#L130) function, drop utils dep and use multihash directly.